File: lab_5/lab_5_cs/lab_5_cs/Methods.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab_5_cs
{
    internal static class Methods
    {
        public static int BiggestVolume(TCylinder[] cylinders)
        {
            int index = 0;
            double max = cylinders[index].FindVolume();
            for (int i = 0; i < cylinders.Length; i++)
            {
                if (cylinders[i].FindVolume() > max)
                {
                    max = cylinders[i].FindVolume();
                    index = i;
                }
            }
            return index;
        }

        public static int LeastSurfaceArea(TPyramid[] pyramids)
        {
            int index = 0;
            double min = pyramids[index].FindVolume();
            for (int i = 0; i < pyramids.Length; i++)
            {
                if (pyramids[i].FindVolume() < min)
                {
                    min = pyramids[i].FindVolume();
                    index = i;
                }
            }
            return index;
        }

        public static void MakePyramidsArray(TPyramid[] pyramids)
        {         
            for (int i = 0; i < pyramids.Length; i++)
            {
                do
                {
                    Console.WriteLine($"\n{i + 1} піраміда:");

                    pyramids[i] = new TPyramid();

                    Console.Write("Висота піраміди: ");
                    double height = double.Parse(Console.ReadLine());
                    height = CheckValue(height);
                    pyramids[i].Height = height;

                    Console.Write("Апофема: ");
                    double apopheme = double.Parse(Console.ReadLine());
                    apopheme = CheckValue(apopheme);
                    pyramids[i].Apopheme = apopheme;

                    Console.Write("Сторорна основи: ");
                    double baseSide = double.Parse(Console.ReadLine());
                    baseSide = CheckValue(baseSide);
                    pyramids[i].BaseSide = baseSide;

                    if (!pyramids[i].CheckData())
                    {
                        Console.WriteLine("-----Некоректні дані! Спробуйте знову.-----");
                    }
                }
                while (!pyramids[i].CheckData());
            }
        }

        public static void MakeCylinderArray(TCylinder[] cylinders)
        {
            for (int i = 0; i < cylinders.Length; i++)
            {
                Console.WriteLine($"\n{i + 1} циліндр:");

                cylinders[i] = new TCylinder();
                
                Console.Write("Радіус: ");
                double radius = double.Parse(Console.ReadLine());
                radius = CheckValue(radius);
                cylinders[i].Radius = radius;

                Console.Write("Висота циліндра: ");
                double height = double.Parse(Console.ReadLine());
                height = CheckValue(height);
                cylinders[i].Height = height;
            }
        }

        public static double CheckValue(double value)
        {
            while(value < 0)
            {
                Console.WriteLine("Неправильно! Число має бути > 0");
                Console.Write("Спробуйте ще раз: ");
                value = double.Parse(Console.ReadLine());
            }
            return value;
        }
    }
}

File: lab_5/lab_5_py/lab_5_py/methods.py
from TPyramid import *
from TCylinder import *

def biggest_volume(cylinders):
    index = 0
    maximum = cylinders[index].find_volume()
    for i in range(len(cylinders)):
        if (cylinders[i].find_volume() > maximum):
            maximum = cylinders[i].find_volume()
            index = i
    return index

def least_surface_area(pyramids):
    index = 0
    minimum = pyramids[index].find_volume()
    for i in range(len(pyramids)):
        if (pyramids[i].find_volume() < minimum):
            minimum = pyramids[i].find_volume()
            index = i
    return index

def check_value(value):
    while value < 0:
        print("Неправильно! Число має бути > 0")
        value = float(input("Спробуйте ще раз: "))
    return value

def make_pyramids_array(pyramids, size):
    for i in range(size):
        while True:
            print(f"\n{i + 1} піраміда:")
            pyramid = []
        
            height = float(input("Висота піраміди: "))
            height = check_value(height)
            pyramid.append(height)

            apopheme = float(input("Апофема: "))
            apopheme = check_value(apopheme)
            pyramid.append(apopheme)

            base_side = float(input("Сторорна основи: "))
            base_side = check_value(base_side)
            pyramid.append(base_side)

            pyramids.append(TPyramid(pyramid))
            if pyramids[i].check_data() == False:
                print("-----Некоректні дані! Спробуйте знову.-----")
            if pyramids[i].check_data() == True:
                break
    
    return pyramids

def make_cylinder_array(cylinders, size):
    for i in range(size):
        print(f"\n{i + 1} циліндр:")
        cylinder = []

        radius = float(input("Радіус: "))
        radius = check_value(radius)
        cylinder.append(radius)

        height = float(input("Висота циліндра: "))
        height = check_value(height)
        cylinder.append(height)

        cylinders.append(TCylinder(cylinder))
    return cylinders